pkg/kubectl/cmd/create_job.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/api/batch/v1beta1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/batch/internalversion"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/util/i18n"
+)
+
+var (
+	jobLong = templates.LongDesc(i18n.T(`
+		Create a Job to execute immediately from a specified CronJob.`))
```

Comment:
Since you're creating `kubectl create job`, which is fine, because I was also thinking about doing it, you need to:

1. make sure it creates just a job, by default
2. only then adding `--from-cronjob` flag.

Reply:
For one see how `kubectl create deployment` works, using a generator and do something similar, and should limit the amount of code needed to write it.

Reply:
I didn't implement a generic `create job` command because I considered it to be out of scope. I'm happy to do so, but maybe it should be part of a separate PR to avoid blowing the size of this one up even more?

Reply:
Separate PR sgtm.